scvi/module/_mixupvae.py

```diff
@@ -479,14 +515,16 @@ def generative(
             size_factor = library
             size_factor_pseudobulk = library_pseudobulk
 
+        one_hot_y = one_hot(y, self.n_labels)
         px_scale, px_r, px_rate, px_dropout = self.decoder(
             self.dispersion,
             decoder_input,
             size_factor,
-            batch_index,
+            one_hot_batch_index,
             *categorical_input,
-            y,
+            one_hot_y,
         )
+        one_hot_y_pseudobulk = one_hot_y[pseudobulk_indices, :].mean(axis=1)
         (
             px_pseudobulk_scale,
```

Comment:
suggestion: we can add an argument like `reconstruct_pseudobulk` set to `False` by default. There's no need to run the decoder on the pseudobulk if we don't add to the loss. Or we can use the `loss_computation` attribute of the class directly

Reply:
The issue is that there are a lot of if statements using the pseudobulk, so needing another if statements makes the code way harder to read, because we have to add them a lot of time